eldritch/test_combat.py

```diff
@@ -1978,5 +1978,108 @@ def testDeactivatesOnCombatLoss(self):
     self.assertTrue(witch.has_attribute("magical resistance", self.state, self.char))
 
 
+class BindMonsterTest(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.bind_monster = items.BindMonster(0)
+    self.char.possessions = [self.bind_monster]
+
+  def start(self, monster):
+    # pylint: disable=attribute-defined-outside-init
+    self.combat = Combat(self.char, monster)
+    self.state.event_stack.append(self.combat)
+
+    with mock.patch.object(
+        events.random, "randint", new=mock.MagicMock(return_value=5)
+    ):
+      fight_or_flee = self.resolve_to_choice(MultipleChoice)
+      fight_or_flee.resolve(self.state, "Fight")
+
+      return self.resolve_to_choice(CombatChoice)
+
+  def testSanePass(self):
+    self.char.sanity = 4
+    self.char.fight_will_slider = 0
+    worm = monsters.GiantWorm()
+    self.start(worm)
+    self.assertEqual(self.char.sanity, 3)
+    self.assertEqual(self.char.stamina, 5)
+    self.state.event_stack.append(self.state.usables[0]["Bind Monster0"])
+    choose_ignore = self.resolve_to_choice(SpendChoice)
+    self.spend("sanity", 2, choose_ignore)
+    choice = self.resolve_to_choice(SpendChoice)
+    choice.resolve(self.state, "Cast")
+    with mock.patch.object(
+        events.random, "randint", new=mock.MagicMock(return_value=5)
+    ):
+      choice = self.resolve_to_choice(CombatChoice)
+
+    choice.resolve(self.state, "done")
+    self.resolve_until_done()
+    self.assertIn(worm, self.char.trophies)
+    self.assertEqual(self.char.sanity, 1)
+    self.assertEqual(self.char.stamina, 4)  # overwhelming
+    self.assertNotIn(self.bind_monster, self.char.possessions)
+
+  def testInsanePass(self):
+    self.char.sanity = 3
+    self.char.fight_will_slider = 0
+    worm = monsters.GiantWorm()
+    self.start(worm)
+    self.assertEqual(self.char.sanity, 2)  # Lost one to nightmarish
+    self.assertEqual(self.char.stamina, 5)
+    self.state.event_stack.append(self.state.usables[0]["Bind Monster0"])
+    choose_ignore = self.resolve_to_choice(SpendChoice)
+    self.spend("sanity", 2, choose_ignore)
+    choice = self.resolve_to_choice(SpendChoice)
```

Comment:
I think you shouldn't need the second `resolve_to_choice(SpendChoice)` anymore. I think `self.spend` takes care of that for you.